src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -72,6 +73,11 @@ public void onReceiveValue(String result) {
 		this.simprints = new SimprintsSupport(this);
 		this.photoGrabber = new PhotoGrabber(this);
 		this.mrdt = new MrdtSupport(this);
+		try {
+			this.smsSender = new SmsSender(this);
+		} catch(Exception ex) {
+			warn(ex, "Failed to create SmsSender.");
```

Comment:
I imagine this would fail on a tablet which doesn't have any SIM card slots, or if the `BroadcastReceiver` fails to register.  If it fails, it's likely not something that can be resolved.  Perhaps it should be logged at `ERROR` level?  Also it may be useful to add e.g. `MedicAndroidFeatureUnsupportedException`, or a `medicmobile_android.sms_available()` method to communicate or detect more clearly (from Javascript) when sms sending isn't going to work.  The latter would be more in-line with other features (e.g. `medicmobile_android.simprints_available()`)